### DISCUSSION TITLE
How to select models "on the fly"?

### DISCUSSION THREAD

**Discussion by ThadeuFerreira** [spring-projects/spring-ai#824] [Q&A]
I've just started with this library and I'm trying to find a way to invoke different models depending on the situation.

Following the tutorials, I have this in my application.properties

```
spring.ai.openai.embedding.options.model=text-embedding-3-small
spring.ai.openai.chat.options.model=gpt-3.5-turbo-0125
...
```

And this configuration

```
    @Bean
    ChatClient chatClient(ChatClient.Builder builder) {
        return builder.build();
    }
```

I couldn't find a way to change the configurations of ChatClient in Running time.
Should I have multiple Beans for each model?
O can I change it in the Controller?

**Comment by abel533** [comment]
```java
OpenAiChatOptions options = OpenAiChatOptions.builder().withModel("model-name").build();
Prompt prompt = new Prompt(new UserMessage(message), options);
return Map.of("generation", chatModel.call(prompt));
```

**Comment by ThomasVitale** [comment]
Using the ChatClient fluent API:

```java
chatClient
        .prompt()
        .options(OpenAiChatOptions.builder().withModel("gtp-4-turbo").build())
        .user(question)
        .call()
        .content();
```

**Reply by abel533** [reply]
I saw the email and thought it was issues pushed by 'cloud-native-spring-in-action'. Upon closer inspection, I realized it was from Spring AI. What a coincidence, 'cloud-native-spring-in-action' was the first book I purchased on Manning.

**Reply by ThomasVitale** [reply]
@abel533 thanks for sharing that. I hope you found the book useful!
